projection/chainstats/chainstats.go

```diff
@@ -0,0 +1,158 @@
+	if getErr != nil {
+		return fmt.Errorf("error getting total block count on init: %v", getErr)
+	}
+	var ok bool
+	if projection.maybeTotalBlockCount, ok = new(big.Int).SetString(rawTotalBlockCount, 10); !ok {
+		return errors.New("error parsing total block count as big.Int on init")
+	}
+
+	return nil
+}
+
+func (projection *ChainStats) HandleEvents(height int64, events []event_entity.Event) error {
+	rdbTx, err := projection.rdbConn.Begin()
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %v", err)
+	}
+
+	committed := false
+	defer func() {
+		if !committed {
+			_ = rdbTx.Rollback()
+		}
+	}()
+
+	rdbTxHandle := rdbTx.ToHandle()
+	chainStatsView := view.NewChainStats(rdbTxHandle)
+
+	event := events[0]
+	var maybeGenesisBlockTime *int64
+	if genesisCreatedEvent, ok := event.(*event_usecase.GenesisCreated); ok {
+		genesisBlockTime, parseErr := utctime.Parse(
+			"2006-01-02T15:04:05.000000Z", genesisCreatedEvent.Genesis.GenesisTime,
+		)
+		if parseErr != nil {
+			return fmt.Errorf("error pasring genesis block time: %v", parseErr)
+		}
+
+		genesisBlockTimeUnixNano := genesisBlockTime.UnixNano()
+		maybeGenesisBlockTime = &genesisBlockTimeUnixNano
+
+		var setErr error
+		if setErr = chainStatsView.Set(
+			GENESIS_BLOCK_TIME, strconv.FormatInt(genesisBlockTime.UnixNano(), 10),
+		); setErr != nil {
+			return fmt.Errorf("error setting genesis block time: %v", setErr)
+		}
+		if setErr = chainStatsView.Set(TOTAL_BLOCK_TIME, "0"); setErr != nil {
+			return fmt.Errorf("error setting total block time: %v", setErr)
+		}
+		if setErr = chainStatsView.Set(TOTAL_BLOCK_COUNT, "0"); setErr != nil {
+			return fmt.Errorf("error setting total block count: %v", setErr)
+		}
+	} else if blockCreatedEvent, ok := event.(*event_usecase.BlockCreated); ok {
+		blockTime := blockCreatedEvent.Block.Time
+
+		totalBlockTime := blockTime.UnixNano() - *projection.maybeGenesisBlockTime
+		totalBlockCount := new(big.Int).Add(projection.maybeTotalBlockCount, big.NewInt(1))
+
+		var setErr error
+		if setErr = chainStatsView.Set(TOTAL_BLOCK_TIME, strconv.FormatInt(totalBlockTime, 10)); setErr != nil {
+			return fmt.Errorf("error setting total block time: %v", setErr)
+		}
+		if setErr = chainStatsView.Set(TOTAL_BLOCK_COUNT, totalBlockCount.String()); setErr != nil {
+			return fmt.Errorf("error setting total block count: %v", setErr)
+		}
+	}
+
+	if err = projection.UpdateLastHandledEventHeight(rdbTxHandle, height); err != nil {
+		return fmt.Errorf("error updating last handled event height: %v", err)
+	}
+
+	if err = rdbTx.Commit(); err != nil {
+		return fmt.Errorf("error committing changes: %v", err)
+	}
+	committed = true
+
+	if maybeGenesisBlockTime != nil {
+		projection.maybeGenesisBlockTime = maybeGenesisBlockTime
+		projection.maybeTotalBlockCount = big.NewInt(0)
+	} else {
+		projection.maybeTotalBlockCount = new(big.Int).Add(projection.maybeTotalBlockCount, big.NewInt(1))
```

Comment:
any chance `projection.maybeTotalBlockCount` could be `nil`, if yes will it misbehave here while doing `.Add`

Reply:
Good question, I have specifically handled the initialization and this value is set on two cases
- OnInit (for cases where some blocks is processed)
- If not, it is first initialized and after processing the genesis block it will be set.

So there's no way it is nil